controllers/baremetalset_controller.go

```diff
@@ -535,27 +573,40 @@ func (r *BaremetalSetReconciler) deleteOwnerRefLabeledObjects(instance *ospdirec
 	return nil
 }
 
-// Get the IP from the management network for a particular BaremetalHost in the BaremetalSet
-func findNewMgmtIP(instance *ospdirectorv1beta1.BaremetalSet, bmhName string) (string, error) {
-	// First check if we already have an IP for this host
-	if bmhStatus, ok := instance.Status.BaremetalHosts[bmhName]; ok {
-		if bmhStatus.MgmtIP != "" {
-			return bmhStatus.MgmtIP, nil
-		}
+// Get the IP, gateway and netmask from the management network for a particular BaremetalHost in the BaremetalSet
+func findIPs(instance *ospdirectorv1beta1.BaremetalSet, bmhName string) (ip string, gateway string, netmask string, err error) {
```

Comment:
maybe we return an "ipset" instead of 3 separate, but as this will change with the discussed network CRDs its ok